### PR TITLE
Samples/Notebooks - Added basic component sharing notebooks

### PR DESCRIPTION
**Pull request by Ark-kun**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/238)

### CODE REVIEW

File: samples/notebooks/Lightweight Python components - sharing 1 - produce component files.ipynb [hunk 1]
```diff
@@ -0,0 +1,155 @@
+{
```
Comment (qimingj) [inline samples/notebooks/Lightweight Python components - sharing 1 - produce component files.ipynb]: It seems neither notebook is big --- merge them? How to produce and how to load are very related.

File: samples/notebooks/Lightweight Python components - sharing 1 - produce component files.ipynb [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Install the SDK\n",
+    "!pip3 install https://storage.googleapis.com/ml-pipeline/release/0.1.3-rc.3/kfp.tar.gz --upgrade\n"
```
Comment (qimingj) [inline samples/notebooks/Lightweight Python components - sharing 1 - produce component files.ipynb]: 0.1.3-rc3 --> 0.1.6. (sorry it has been a while after this change is out for review).

File: samples/notebooks/Lightweight Python components - sharing 1 - produce component files.ipynb [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    "\n",
+    "Lightweight python components do not require you to build a new container image for every code change.\n",
+    "They're intended to use for fast iteration in notebook environment.\n",
+    "\n",
+    "#### Building a lightweight python component\n",
+    "To build a component just define a stand-alone python function and then call kfp.components.func_to_container_op(func) to convert it to a component that can be used in a pipeline.\n",
+    "\n",
+    "There are several requirements for the function:\n",
+    "* The function should be stand-alone. It should not use any code declared outside of the function definition. Any imports should be added inside the main function. Any helper functions should also be defined inside the main function.\n",
+    "* The function can only import packages that are available in the base image. If you need to import a package that's not available you can try to find a container image that already includes the required packages. (As a workaround you can use the module subprocess to run pip install for the required package.)\n",
```
Comment (qimingj) [inline samples/notebooks/Lightweight Python components - sharing 1 - produce component files.ipynb]: Maybe add a link to the sample notebook which demonstrates how to build a container inside cluster? Or even include some code snippets?

File: samples/notebooks/Lightweight Python components - sharing 1 - produce component files.ipynb [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+   "outputs": [],
+   "source": [
+    "import kfp.components as comp"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Simple function that just add two numbers:"
```
Comment (qimingj) [inline samples/notebooks/Lightweight Python components - sharing 1 - produce component files.ipynb]: add --> adds.

File: samples/notebooks/Lightweight Python components - sharing 2 - consume component file.ipynb [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import kfp.dsl as dsl\n",
+    "@dsl.pipeline(\n",
+    "   name='Calculation pipeline',\n",
+    "   description='A toy pipeline that performs arithmetic calculations.'\n",
+    ")\n",
+    "def calc_pipeline(\n",
+    "   a=dsl.PipelineParam('a'),\n",
```
Comment (qimingj) [inline samples/notebooks/Lightweight Python components - sharing 2 - consume component file.ipynb]: It doesn't have to be "dsl.PipelineParam" type now.

**Comment by vicaire** [comment]
@Ark-kun, do you still need this one? Should we close?

**Comment by Ark-kun** [comment]
> @Ark-kun, do you still need this one? Should we close?

I think so. We almost released this before GA, but then decided to release it right after GA.

**Comment by gaoning777** [comment]
What is the status of this PR? Should we close it or it's good for review?

**Pull request closed without merging**